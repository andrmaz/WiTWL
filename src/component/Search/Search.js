import React from 'react';
import PropTypes from 'prop-types'; // ES6
import './Search.scss';

const Search = ({ city, handleChange, handleSubmit }) => {
    return (
        <div>
            <form onSubmit={handleSubmit}>
                <label>
                    City :
                    <input 
                        id='q'
                        required
                        pattern='^[a-zA-Z]+$'
                        title='Search by city'
                        type="text" 
                        name="city" 
                        value={city}
                        onChange={handleChange}
                    />
                </label>
                <input type="submit" value="Submit"/>
            </form>
        </div>
    )
}

Search.propTypes = {
    handleSubmit: PropTypes.func,
    city: PropTypes.string,
    handleChange: PropTypes.func,
}

export default Search;
