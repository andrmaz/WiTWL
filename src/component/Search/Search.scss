.box1 {
    display: flex;
    justify-content: center;
    align-items: center;

    #q {
        border-radius: 3px;
        padding: 2px;

        &:valid {
            border: 3px solid green;
        }
        &:invalid {
            border: 2px solid red;
        }
    }

    .label {
        font-weight: bold;
        font-size: 18px;
    }

    
    #button {
        cursor: pointer;
        border-radius: 3px;
        padding: 2px;
        background-color: black;
        color: white;
        
        &:focus {
            transform: scale(0.9);
            transition: width 2s linear 1s;
        }
    }
}

@media screen and (max-width: 1024px) {
    .box1 {
        grid-row-start: 1;
        grid-row-end: 2;
        grid-column-start: 1;
        grid-column-end: 3;
    }
}