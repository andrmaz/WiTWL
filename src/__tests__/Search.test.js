import React from 'react';
import { shallow } from 'enzyme';
import Search from '../component/Search/Search';

describe('Search component', () => {
    it('should render component correctly', () => {
        const wrapper = shallow(<Search />);
        expect(wrapper).toMatchSnapshot()
    })
})