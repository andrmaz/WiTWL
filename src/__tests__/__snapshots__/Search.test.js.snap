// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render component correctly 1`] = `
<div
  className="box1"
>
  <form>
    <label
      className="label"
    >
      City : Â 
      <input
        id="q"
        maxLength="20"
        name="city"
        pattern="^([a-zA-Z]+\\\\s)*[a-zA-Z]+$"
        required={true}
        size="30"
        title="Please make sure to remove unused white space"
        type="text"
      />
    </label>
    <input
      id="button"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
